### ISSUE TITLE
flow_from_dictionary cannot produce the same list of augmented data with fixed seed

### ISSUE THREAD

**Issue by zhenglilei** [keras-team/keras#10169]
The findings are simple: for Keras-gpu=2.0.8, we can have the same list of augmented data generated with fixed seed (e.g., 1); for the later versions, > 2.1.2, the generated list is somewhat random even the seed is fixed. 

train_datagen = image.ImageDataGenerator(
        rescale=1./255,
        rotation_range=0,
        width_shift_range=0.2,
        height_shift_range=0.2,
        shear_range=0.2,
        zoom_range=0.2,
        horizontal_flip=True,
        vertical_flip = True,
        fill_mode='nearest')

train_generator = train_datagen.flow_from_directory(
        train_data_dir,
        target_size=(img_height, img_width),
        batch_size=100,
        class_mode='binary',
        save_to_dir='augmentedData',
        seed=1)

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by zhenglilei** [comment]
https://github.com/zhenglilei/Keras_data_order_in_batch.git

Please see the codes to reproduce the findings in the above link.

**Comment by zhenglilei** [comment]
I checked the keras codes and found that the problem is with the "fit_generator" function. The "fit_generator"  function calls the "OrderedEnqueuer" class to shuffle the sequence data generated by the "flow_from_dictionary" function. However, before shuffling in the "OrderedEnqueuer/_run" function , no random seed is specified. Although there is no need to have a 'seed' parameter in the call of the "fit_generator", we can still use the seed which is specified in the "flow_from_dictionary" function.

In the "data_utils.py" file, Class "OrderedEnqueuer", function "_run", add the three line codes:
![image](https://user-images.githubusercontent.com/8134656/41718196-e1a3292c-758e-11e8-932f-62baddec4791.png)

**Comment by zhenglilei** [comment]
Since this issue has been cleaned, I close it.
